uapi/include/uapi/types.h

```diff
@@ -301,6 +302,22 @@ typedef struct shm_infos {
     uint32_t perms;   /*< SHM permissions (mask of SHMPermission) */
 } shm_infos_t;
 
+/**
+ * @brief generic state value definition for a DMA stream, used to get back DMA statuses
+ */
+typedef enum dma_chan_state {
+    GPDMA_STATE_IDLE                    = 1, /**< DMA channel idle (not set or unused) */
+    GPDMA_STATE_RUNNING                 = 2, /**< DMA channel is running */
+    GPDMA_STATE_ABORTED                 = 3, /**< DMA stream aborted on SW request */
+    GPDMA_STATE_SUSPENDED               = 4, /**< DMA stream suspended on SW request*/
+    GPDMA_STATE_TRANSMISSION_FAILURE    = 5, /**< DMA transmission failure */
+    GPDMA_STATE_CONFIGURATION_FAILURE   = 6, /**< DMA channel configuration failure */
+    GPDMA_STATE_OVERRUN                 = 7, /**< DMA transmission overrun */
+    GPDMA_STATE_TRANSFER_COMPLETE       = 8, /**< DMA transfer complete for this channel */
```

Comment:
Yes, these points are manipulated at interrupt event level (i.e. the IRQn handler will read back the current status from the gpdma driver). I'm adding that in another PR to avoid loong PR :)